task 2: ast rewrite

migeed-z opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61310
* #61265
* #61239
* #61238
* #61188
* #61150
* #61093
* #61012
* #60805
* **#60622**

Differential Revision: [D29493747](https://our.internmc.facebook.com/intern/diff/D29493747)

migeed-z committed:
task 2: ast rewrite

[ghstack-poisoned]

migeed-z committed:
Update on "task 2: ast rewrite"




[ghstack-poisoned]

migeed-z committed:
Update on "task 2: ast rewrite"




[ghstack-poisoned]

migeed-z committed:
Update on "task 2: ast rewrite"




[ghstack-poisoned]

migeed-z committed:
Update on "task 2: ast rewrite"




[ghstack-poisoned]

migeed-z committed:
Update on "task 2: ast rewrite"




[ghstack-poisoned]

migeed-z committed:
Update on "task 2: ast rewrite"




[ghstack-poisoned]

jamesr66a commented on torch/fx/experimental/rewriter.py:
```diff
@@ -64,6 +64,18 @@ def visit_Assert(self, node):
 
+    def visit_AnnAssign(self, node):
+        """
+        Swap out Python's AnnAssign with an Assign note where the annotation function is called.
```
nit: `s/note/node`

jamesr66a reviewed (approved):
![image](https://user-images.githubusercontent.com/4685384/123713194-b6143080-d828-11eb-9f1a-77ac109ad68f.png)

migeed-z committed:
Update on "task 2: ast rewrite"




[ghstack-poisoned]

migeed-z committed:
Update on "task 2: ast rewrite"




[ghstack-poisoned]

migeed-z commented:
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29493747).

migeed-z committed:
Update on "task 2: ast rewrite"


Differential Revision: [D29493747](https://our.internmc.facebook.com/intern/diff/D29493747)

[ghstack-poisoned]

migeed-z commented:
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29493747).

migeed-z committed:
Update on "task 2: ast rewrite"


Differential Revision: [D29493747](https://our.internmc.facebook.com/intern/diff/D29493747)

[ghstack-poisoned]
